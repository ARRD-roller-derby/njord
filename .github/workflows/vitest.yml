name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30