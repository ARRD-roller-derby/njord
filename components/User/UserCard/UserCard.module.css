.container {
  background: var(--color-bg-light-extra);
  border-radius: 10px;
  display: grid;
  grid-template-columns: auto 2fr;
  cursor: pointer;
  border: 1px solid transparent;
  transition: 200ms ease-in-out;
  overflow: hidden;
  flex-basis: 350px;
}

.container:hover {
  border: 1px solid var(--color-arrd);
}

.avatarContainer {
  position: relative;
  display: flex;
  align-content: center;
  align-items: center;
  width: 70px;
  padding: 8px 0 8px 10px;
}

.avatar {
  width: 50px;
  height: 50px;
  z-index: 2;
}

.avatarContainer:before {
  content: "";
  position: absolute;
  top: 0;
  right: calc(50% - 5px);
  left: 0;
  bottom: 0;
  border-radius: 10px 0 0 10px;
  background: var(--color-arrd);
}

.identity {
  display: flex;
  flex-direction: column;
  padding: 10px;
  gap: 5px;
  color: var(--color-txt-light);
  justify-content: space-between;
}

.nameContainer {
  font-size: 12px;
}

.name::first-letter,
.lastname,
.derbyName::first-letter {
  text-transform: uppercase;
}

.derbynameContainer {
  display: flex;
  gap: 5px;
  align-items: center;
  font-size: 12px;
  color: var(--color-txt);
}

.icon {
  color: var(--color-tierce);
}

.teams {
  display: flex;
  gap: 5px;
  flex-wrap: wrap;
  font-size: 10px;
  justify-content: left;
  color: var(--color-tierce);
}
