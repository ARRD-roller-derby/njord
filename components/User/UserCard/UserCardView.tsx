import { UserInterface } from '../../../types/User.interface'
import Avatar from '../../_ui/Avatar/Avatar'
import classes from './UserCard.module.css'

interface props {
  readonly user: UserInterface
  readonly handleClick: Function
}

export default function UserCardView({ user, handleClick }: props) {
  return (
    <div className={classes.container}>
      <div className={classes.avatarContainer}>
        <div className={classes.avatar}>
          <Avatar src={user.avatar} />
        </div>
      </div>
      {/** Use Shallow for change url without redirect. if reload, is effect. */}
      <div className={classes.identity} onClick={() => handleClick()}>
        <div className={classes.nameContainer}>
          <span className={classes.name}>{user.name || user.email}</span>{' '}
          <span className={classes.lastname}>{user.lastname}</span>
        </div>

        {user.derbyName && (
          <div className={classes.derbynameContainer}>
            {'#'}
            <span className={classes.numRoster}>{user.numRoster}</span>{' '}
            <span className={classes.derbyName}>{user.derbyName}</span>
          </div>
        )}
        <div className={classes.teams}>
          {user.teams.map((team) => (
            <div key={team} className={classes.team}>
              {team}
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}
