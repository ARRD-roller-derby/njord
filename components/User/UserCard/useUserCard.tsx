import { UserInterface } from '../../../types/User.interface'
import { useRouter } from 'next/router'

export default function useUserCard(user: UserInterface, openPopin: Function, url:string) {
  const router = useRouter()

  function handleClick() {
    openPopin(user)
    router.push(url, '/user/' + user._id, { shallow: true })
  }

  return handleClick
}
