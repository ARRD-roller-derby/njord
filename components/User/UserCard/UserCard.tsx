import { UserInterface } from '../../../types/User.interface'
import UserCardView from './UserCardView';
import useUserCard from './useUserCard';

interface props {
  readonly user: UserInterface
  readonly openPopin: Function
  readonly url: string
}

export default function UserCard({ user, openPopin, url }: props) {
  const handleClick = useUserCard(user,openPopin, url);
  
  return <UserCardView user={user} handleClick={handleClick}/>
}
