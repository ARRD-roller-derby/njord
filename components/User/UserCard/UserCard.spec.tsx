import { afterEach, describe, expect, it, vi } from 'vitest'
import { render, cleanup, renderHook, act } from '@testing-library/react'
import UserCard from './UserCard'
import useUserCard from './useUserCard'

describe('<UserCard />', () => {
  afterEach(cleanup)

  const mockUser = {
    _id: 'id',
    email: 'on@ok.fr',
    name: 'name',
    lastname: 'lastname',
    leagues: ['league', 'league 2'],
    teams:[]
  }

  it('Check snapshot', () => {
    const { asFragment } = render(
      <UserCard user={mockUser} openPopin={vi.fn} url='url'/>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Test handleClick', () => {
    const triggerOpenPopin = vi.fn(),
      { result } = renderHook(() => useUserCard(mockUser, triggerOpenPopin, 'url'))

    act(() => {
      result.current()
    })

    expect(triggerOpenPopin).toBeCalledWith(mockUser)
  })
})