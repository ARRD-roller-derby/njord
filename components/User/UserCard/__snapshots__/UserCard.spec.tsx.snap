// Vitest Snapshot v1

exports[`<UserCard /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_r7wl2_1"
  >
    <div
      class="_avatarContainer_r7wl2_17"
    >
      <div
        class="_avatar_r7wl2_17"
      >
        <div
          class="_avatar_1yrdp_1"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              alt="avatar"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
    <div
      class="_identity_r7wl2_43"
    >
      <div
        class="_nameContainer_r7wl2_52"
      >
        <span
          class="_name_r7wl2_52"
        >
          name
        </span>
         
        <span
          class="_lastname_r7wl2_57"
        >
          lastname
        </span>
      </div>
      <div
        class="_teams_r7wl2_74"
      />
    </div>
  </div>
</DocumentFragment>
`;
